src/Common/src/Interop/Unix/System.Native/Interop.ReadDir.cs

```diff
@@ -50,14 +50,17 @@ internal ReadOnlySpan<char> GetName(Span<char> buffer)
 
                 int charCount = Encoding.UTF8.GetChars(nameBytes, buffer);
                 ReadOnlySpan<char> value = buffer.Slice(0, charCount);
-                Debug.Assert(value.IndexOf('\0') == -1, "should not have embedded nulls");
+                Debug.Assert(NameLength != -1 || value.IndexOf('\0') == -1, "should not have embedded nulls if we parsed the end of string");
```

Comment:
parens would help on
`ReadOnlySpan<byte> nameBytes = NameLength == -1`

Reply:
Nit I would find this easier to read
`Debug.Assert(!(NameLength == -1 && value.IndexOf('\0') != -1), "should not have embedded nulls if we parsed the end of string");`

Reply:
Ok

Reply:
I find the assert alternative harder to read myself. Adding the parens (your second comment wasn't showing when I responded)

Reply:
ok